package.json

```diff
@@ -129,7 +129,7 @@
     "@hapi/boom": "^9.1.4",
     "@hapi/cookie": "^11.0.2",
     "@hapi/h2o2": "^9.1.0",
-    "@hapi/hapi": "^20.2.2",
+    "@hapi/hapi": "21.3.2",
```

Comment:
Based on the diff you shared I think it would break this test https://github.com/elastic/kibana/blob/main/src/core/server/integration_tests/http/http_server.test.ts#L119-L122

The impact on users would be that they receive a 502 without any message instead of a 503 with "Kibana is shutting down and not accepting new incoming requests". While it was nice that we were giving a descriptive error back I don't think this would be considered a breaking change.

So I think it would be better to adopt the new version and change our tests accordingly.

Reply:
Sounds good to me! I'll push this PR forward then :)